game_management/game_input_handler.py

```diff
@@ -0,0 +1,593 @@
+                                abs(unit.location[1] - data.settlement.location[1]) <= 1:
+                            unit.besieging = False
+                    # The Concentrated can only have a single settlement, so when they take others, the
+                    # settlements simply disappear.
+                    if game_state.players[0].faction is not Faction.CONCENTRATED:
+                        game_state.players[0].settlements.append(data.settlement)
+                    for idx, p in enumerate(game_state.players):
+                        if data.settlement in p.settlements and idx != 0:
+                            p.settlements.remove(data.settlement)
+                            break
+                game_state.board.overlay.toggle_setl_attack(data)
+                game_state.board.attack_time_bank = 0
+            case SettlementAttackType.BESIEGE:
+                # Alternatively, begin a siege on the settlement.
+                game_state.board.selected_unit.besieging = True
+                game_state.board.overlay.attacked_settlement.besieged = True
+                game_state.board.overlay.toggle_setl_click(None, None)
+            case _:
+                game_state.board.overlay.toggle_setl_click(None, None)
+    elif game_state.game_started and game_state.board.overlay.is_pause():
+        match game_state.board.overlay.pause_option:
+            case PauseOption.RESUME:
+                game_state.board.overlay.toggle_pause()
+            case PauseOption.SAVE:
+                save_game(game_state)
+                game_state.board.overlay.has_saved = True
+            case PauseOption.CONTROLS:
+                game_state.board.overlay.show_additional_controls = False
+                game_state.board.overlay.toggle_controls()
+            case PauseOption.QUIT:
+                game_state.game_started = False
+                game_state.on_menu = True
+                game_controller.menu.loading_game = False
+                game_controller.menu.in_game_setup = False
+                game_controller.menu.main_menu_option = MainMenuOption.NEW_GAME
+                game_controller.music_player.stop_game_music()
+                game_controller.music_player.play_menu_music()
+    elif game_state.game_started and not (
+            game_state.board.overlay.is_tutorial() or game_state.board.overlay.is_deployment() or
+            game_state.board.overlay.is_bless_notif() or
+            game_state.board.overlay.is_constr_notif() or game_state.board.overlay.is_lvl_notif() or
+            game_state.board.overlay.is_close_to_vic() or
+            game_state.board.overlay.is_investigation() or game_state.board.overlay.is_night()):
+        # If we are not in any of the above situations, end the turn.
+        if game_state.end_turn():
+            # Autosave every 10 turns.
+            if game_state.turn % 10 == 0:
+                save_game(game_state, True)
```

Comment:
Probably best to put the `auto` kwarg back in here, for clarity.

Reply:
[588b895](https://github.com/ChrisNeedham24/microcosm/pull/111/commits/588b895112c545f17b9883b616153d3514b540f7)